updated readme (#29)
UBC-MDS/KmeansR: README.Rmd

## Milestone 1 README

This package will include R packages that implement k-means clustering from scratch. This will work on any dataset with valid numerical features, and includes fit, predict, and score functions, as well as as elbow and silhouette methods for hyperparameter “k” optimization.

+ fit: This function classifies the non-labeled data into a given number of clusters k using simple KMeans algorithm. It returns labels for each data point according to the cluster it belongs and also cluster centers. This is a type of unsupervised learning method to classify data.

+ predict: Assigns each point in a dataset to a cluster. Dataset has to be in the same format as the original dataset the model was fit on.

> + score: Outputs a score based on goodness of fit.

Reviewer: Can we add what measure we are using for the goodness of fit

+ elbow: Creates a plot of inertia vs number of cluster centers as per the elbow method. Calculates and returns the inertia values for all cluster centers. Useful for identifying the optimal number of clusters while using k-means clustering algorithm.

> + silhouette: Returns the average silhouette score of each sample in a given 2-d array and clustering labels.

Reviewer: ```suggestion
This function returns the average silhouette score of each sample in a given 2-d array and clustering labels. Silhouette score gives a measure of the separation of the clusters. The higher the silhouette score better is the model
```

+ cluster_summary: Provides summary of groups created from Kmeans clustering, including centroid coordinates, number of data points in training data assigned to each cluster, and within-cluster distance metrics.

> There is a python package sklearn.cluster.KMeans that has similar functions, and a built in k-means function in R. These packages are not meant to add to the existing ecosystem; they are rather intended to deepen fundamental understanding of these algorithms.

Reviewer: ```suggestion
There is an existing built-in k-means function in R. in the `cluster` package. The packages published here are not meant to add to the existing ecosystem; they are rather intended to deepen the fundamental understanding of these algorithms.
```
